WordPress/WordPressTest/BlogJetpackTest.m

```diff
@@ -75,19 +75,17 @@ - (void)testJetpackUsername {
 }
 
 - (void)testJetpackSetupDoesntReplaceDotcomAccount {
-    XCTestExpectation *saveExpectation = [self expectationWithDescription:@"Context save expectation"];
-    self.testContextManager.testExpectation = saveExpectation;
+    XCTestExpectation *saveExpectation = [self expectationForNotification:NSManagedObjectContextDidSaveNotification object:self.testContextManager.mainContext handler:nil];
 
     AccountService *accountService = [[AccountService alloc] initWithManagedObjectContext:[ContextManager sharedInstance].mainContext];
```

Comment:
🤔 it's surprising to see the expectation using `self.textContextManager.mainContext` and this using `[ContextManager sharedInstance].mainContext`.

They look like different contexts. Or am I missing something?

For reference, here's where each is instantiated (I think)

For `ContextManager`:

https://github.com/wordpress-mobile/WordPress-iOS/blob/112b352938ecbea191addd94be7f61d8cd4d5bf4/WordPress/Classes/Utility/ContextManager.m#L72-L83

For `TestContextManager`, which internally uses `ContextManager`:

https://github.com/wordpress-mobile/WordPress-iOS/blob/112b352938ecbea191addd94be7f61d8cd4d5bf4/WordPress/WordPressTest/ContextManagerMock.m#L70-L80

Reply:
Good catch! The code does look like they are referencing different context objects. But at runtime, they are actually the same one, the one in `ContextManagerMock`, since TestContextManager [overrides the shared instance](https://github.com/wordpress-mobile/WordPress-iOS/blob/98dc3b777d8e2cc5f82b66b5e2a2b92153e74e3c/WordPress/WordPressTest/TestContextManager.m#L15-L16)(the overriding actually happens [within `ContextManagerMock`](https://github.com/wordpress-mobile/WordPress-iOS/blob/98dc3b777d8e2cc5f82b66b5e2a2b92153e74e3c/WordPress/WordPressTest/ContextManagerMock.m#L23)). Which is why `self.testContextManager.mainContext` is same as `[[ContextManager sharedInstance] mainContext]`.

I will update the code to use the same reference for clarity.

BTW, the above overriding behaviour will be updated in my establishing usage pattern PR.

Reply:
Gotcha! I knew I must have been missing something, else I would have expected the tests to fail 😄 

Thank you for making it explicit in https://github.com/wordpress-mobile/WordPress-iOS/pull/18485/commits/c6873f2c15b02d4593c31565dac87e2924e85058. 👍